interfaces/IBF-dashboard/src/app/services/timeline.service.ts

```diff
@@ -334,41 +239,40 @@ export class TimelineService {
       ) as LeadTime;
     }
 
-    for (const leadTime of disasterLeadTimes) {
-      // Push first only active lead-times ..
-      if (
-        !visibleLeadTimes.map((lt) => lt.leadTime).includes(leadTime) &&
-        this.isLeadTimeEnabled(leadTime, visibleLeadTimes)
-      ) {
-        if (this.hasDisabledTimeline(this.disasterType.disasterType)) {
-          // for non-interactive timeline: add lead-time only once
+    for (const event of this.eventState.events) {
+      // Push first event starting lead-times ..
+      const duration = event.duration || 1;
+      const startLeadTimeToUse =
+        event.firstTriggerLeadTime || event.firstLeadTime;
+      for (let i = 0; i < duration; i++) {
+        // .. for events with duration (drought) also push the following lead-times
+        const leadTime =
+          `${String(Number(startLeadTimeToUse.split('-')[0]) + i)}-${
+            event.firstLeadTime.split('-')[1]
+          }` as LeadTime;
+        if (!visibleLeadTimes.map((lt) => lt.leadTime).includes(leadTime)) {
           visibleLeadTimes.push({
             leadTime,
-            eventName: null,
-            undefined: false,
+            eventNames: [event.eventName],
+            forecastAlert: true,
+            forecastTrigger: event.forecastTrigger,
+            undefinedLeadTime: false,
```

Comment:
@jannisvisser I'm afraid to ask what an _undefined lead time_ is.

Reply:
haha. Good to know though, it relates directly to the 'no-landfall-yet' scenario in typhoon, where no lead time can be predicted as of yet. You can mock it to get a feel for what it is. Does it make sense to document this somewhere, in terms of specific code being tied to this specific scenario?

Reply:
done